File: 392_IsSubsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
	var i = 0, j = 0;
	if (!s) {
		return true;
	}
    while (j < t.length) {
    	if (s[i] === t[j]) {
    		i += 1;
    	}
    	j += 1;
    	if (i === s.length) {
    		return true;
    	}
    }

    return false;
};

File: 392.IsSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0) {
            return true;
        }
        int indexS = 0;
        int indexT = 0;
        while (indexT < t.length()) {
            if (t.charAt(indexT) == s.charAt(indexS)) {
                indexS += 1;
                if (indexS == s.length()) {
                    return true;
                }
            }
            indexT += 1;
        }
        return false;
    }
}
